File: server/src/main/java/com/fluffy/comment/domain/ExamCommentRepositoryCustom.java
package com.fluffy.comment.domain;

import com.fluffy.comment.domain.dto.ExamReplyCommentDto;
import com.fluffy.comment.domain.dto.ExamRootCommentDto;
import java.util.List;

public interface ExamCommentRepositoryCustom {

    List<ExamRootCommentDto> findRootComments(Long examId);

    List<ExamReplyCommentDto> findRootCommentWithReplies(Long commentId);
}

File: server/src/main/kotlin/com/fluffy/comment/domain/ExamCommentRepositoryCustom.kt
package com.fluffy.comment.domain

import com.fluffy.comment.domain.dto.ExamReplyCommentDto
import com.fluffy.comment.domain.dto.ExamRootCommentDto

interface ExamCommentRepositoryCustom {

    fun findRootComments(examId: Long): List<ExamRootCommentDto>

    fun findReplyComments(commentId: Long): List<ExamReplyCommentDto>
}